lib/spack/spack/solver/asp.py

```diff
@@ -1512,22 +1539,30 @@ def condition(
         Arguments:
             required_spec: the constraints that triggers this condition
             imposed_spec: the constraints that are imposed when this condition is triggered
-            name: name for `required_spec` (required if required_spec is anonymous, ignored if not)
+            required_name: name for ``required_spec``
+                (required if required_spec is anonymous, ignored if not)
+            imposed_name: name for ``imposed_spec``
+                (required if imposed_spec is anonymous, ignored if not)
```

Comment:
Not at the moment, but it's added because this PR introduces a place where there is an anonymous imposed spec via conditional values. There were never previously anonymous imposed specs. 

I don't think it makes sense to assume that if we pass `required_spec` and `required_name`, `imposed_spec` should also use `required_name` if anonymous.  They're two different specs that may come from two different places, so this is there for symmetry.